src/diff.rs

```diff
@@ -939,4 +1149,59 @@ mod tests {
         assert_eq!(stats.deletions(), 0);
         assert_eq!(stats.files_changed(), 0);
     }
+
+    mod foreach {
```

Comment:
I did this as an alternative to mentioning the method name in each test, I like what it does for the `cargo test` output, giving lines like `test diff::tests::foreach::file_and_hunk`. I can pull them out and put a `foreach_` prefix on them if you think that would be better.